### PR TITLE
motdgen: convert to long-running process

### PR DESCRIPTION
**Pull request by crawford**
This will prevent the log from being filled with messages about the
timer starting and stopping. This also changes the process to a notify
process, so that we can ensure that the MOTD is generated before it is
needed.

### CODE REVIEW

**Comment by crawford** [comment]
Fixes https://github.com/coreos/bugs/issues/47.

**Comment by jonboulle** [comment]
can you wrap the stuff from https://github.com/coreos/init/pull/183 in here?

**Comment by crawford** [comment]
Merged and rebased.

File: scripts/motdgen [hunk 1]
```diff
@@ -4,25 +4,31 @@
...
-	echo -e "Failed Units: \033[31m${count}\033[39m" >> /run/coreos/motd
-	awk '{ print "  " $1 }' /run/coreos/motd-failed >> /run/coreos/motd
-fi
+	systemctl list-units --state=failed --no-legend > /run/coreos/motd-failed
+	if [[ -s /run/coreos/motd-failed ]]; then
+		echo -e "Failed Units: \033[31m${count}\033[39m" >> /run/coreos/motd
+		awk '{ print "  " $1 }' /run/coreos/motd-failed >> /run/coreos/motd
+	fi
+
+	systemd-notify --ready
```
Comment (jonboulle) [inline scripts/motdgen]: do you really want to do this on every loop?

Comment (crawford) [inline scripts/motdgen]: :shrug:

File: scripts/motdgen [hunk 2]
```diff
@@ -4,25 +4,31 @@
...
-# We want to grab the GROUP variable
-[ -e /usr/share/coreos/update.conf ] && source /usr/share/coreos/update.conf
-[ -e /etc/coreos/update.conf ] && source /etc/coreos/update.conf
+while [ 1 ]; do
+	# We want to grab the GROUP variable
+	[ -e /usr/share/coreos/update.conf ] && source /usr/share/coreos/update.conf
+	[ -e /etc/coreos/update.conf ] && source /etc/coreos/update.conf
 
-source /usr/lib/os-release
+	source /usr/lib/os-release
```
Comment (marineam) [inline scripts/motdgen]: No point to sourcing this multiple times

File: scripts/motdgen [hunk 3]
```diff
@@ -4,25 +4,31 @@
...
 
-mkdir -p /run/coreos
-echo -e "Core\033[38;5;206mO\033[38;5;45mS\033[39m ${GROUP} (${VERSION})" > /run/coreos/motd
+	mkdir -p /run/coreos
```
Comment (marineam) [inline scripts/motdgen]: mkdir could be done just once

**Comment by vcaputo** [comment]
this is gross

**Comment by marineam** [comment]
I'm leaning towards nope on this. If we are going to convert this tool back into a long running process it needs to be rewritten in a real language. The reason it is on a timer in the first place is because the first implementation attempted to feed off of dbus events but doing so in bash wound up using a lot of CPU.

**Comment by marineam** [comment]
Sure logs are noisy, but so what? They are a reminder that this script is still crap :-P

**Comment by crawford** [comment]
OK, new question: should we bother with the failed units in the MOTD?

**Comment by vcaputo** [comment]
I think we should drop the failed units from motdgen, and stick the systemctl invocation in a profile or bashrc for the core user or something like that instead.  It's arguably inappropriate to show the failed units to all users on login anyways.

**Comment by vcaputo** [comment]
@crawford @marineam PTAL:
https://github.com/coreos/init/pull/187
https://github.com/coreos/baselayout/pull/40

**Pull request closed without merging**